sqldelight-compiler/src/test/kotlin/com/squareup/sqldelight/core/DialectRule.kt

```diff
@@ -0,0 +1,41 @@
+package com.squareup.sqldelight.core
+
+import com.alecstrong.sql.psi.core.DialectPreset
+import com.alecstrong.sql.psi.core.DialectPreset.HSQL
+import com.alecstrong.sql.psi.core.DialectPreset.MYSQL
+import com.alecstrong.sql.psi.core.DialectPreset.POSTGRESQL
+import com.alecstrong.sql.psi.core.DialectPreset.SQLITE_3_18
+import com.alecstrong.sql.psi.core.DialectPreset.SQLITE_3_24
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+annotation class DialectTest(
+  val dialects: Array<DialectPreset>
+)
+
+class DialectRule : TestRule {
+  lateinit var dialect: DialectPreset
```

Comment:
Not super comfortable with this being lateinit. Would rather see a normal nullable value and `!!` in `textType` to more accurately reflect that its value changes over time rather than being set once just late.

Reply:
coolcoolcoolcool

Reply:
actually that also means we have to !! whenever we want to reference it from out of this file, which doesn't accurately reflect the api here since dialect should always be non-null when you're asking for it. I could have a separate backing field for inner which is nullable but that kinda feels like overkill

Reply:
Didn't realize it was public. Can you at least do a `private set` or does the `lateinit` prevent that?

Reply:
nope we can do that! cool